backend/core/src/auth/dto/user-invite.dto.ts

```diff
@@ -0,0 +1,28 @@
+import { OmitType } from "@nestjs/swagger"
+import { UserDto } from "./user.dto"
+import { Expose, Type } from "class-transformer"
+import { IsDefined, IsOptional, ValidateNested } from "class-validator"
+import { ValidationsGroupsEnum } from "../../shared/types/validations-groups-enum"
+import { UserRolesCreateDto } from "./user-roles.dto"
+import { IdDto } from "../../shared/dto/id.dto"
+
+export class UserInviteDto extends OmitType(UserDto, [
+  "id",
+  "createdAt",
+  "updatedAt",
+  "roles",
+  "leasingAgentInListings",
+] as const) {
+  @Expose()
+  @IsOptional()
+  @IsDefined({ groups: [ValidationsGroupsEnum.default] })
+  @Type(() => UserRolesCreateDto)
+  roles?: UserRolesCreateDto | null
```

Comment:
Should roles be optional for the invite? If we don't assign them isPartner, then they won't be able to access the portal.

Reply:
> should roles be required when inviting a user? If we don't assign them any roles, then they won't be able to login

@seanmalbert We don't have any logic in place to prevent partners portal login based on roles, any user can login in (applicants too). 

> I think jurisdiction should also be required and not empty

I have not made it required because in the designs there is no jurisidction input. 

> should the invite endpoint check if the user exists? What happens with what's here and a user with that email already exists?

`invite` reuses `UserService.createUser` which in turn checks that and throws: 
```
  EMAIL_IN_USE: { message: "emailInUse", status: HttpStatus.BAD_REQUEST },
```

Reply:
@seanmalbert Why jurisdictions should be required?

Reply:
@pbn4 ,
This doesn't matter so much for user's assigned with isPartner, since they are directly assigned listings and can't create them, but for admin users, they need to be assigned a jurisdiction so we can know which jurisdictions they can create listings in. The mocks don't show it, but if an admin has access to more than one jurisdiction there needs to be an option to select which jurisdictions the new user has. If the admin is only assigned to one, which will be the case for everyone but us (for now anyway), then the input will be hidden with the value set to the jurisdiction they have. So on the backend with these new endpoints, we also need to ensure that the jurisdictions passed into creating a partner user are limited to what the current user has. I think we already updated the user update function to check this.